internal/adminapi/kong.go

```diff
@@ -16,7 +16,16 @@ import (
 	tlsutil "github.com/kong/kubernetes-ingress-controller/v2/internal/util/tls"
 )
 
+type KongClientNotReadyError struct {
+	Err error
+}
+
+func (e KongClientNotReadyError) Error() string {
+	return fmt.Errorf("client not ready: %w", e.Err).Error()
```

Comment:
Why not like that?
```suggestion
	return fmt.Sprintf("client not ready: %s", e.Err)
```
by the end of the day, it returns a string in both cases